python/GafferTest/NodeTest.py

```diff
@@ -349,5 +349,36 @@ def assertConnections() :
 		s.addChild( n )
 		assertConnections()
 
+	def testReconnectionOfChildPlug( self ) :
```

Comment:
I think this should be in ScriptNodeTest, where all the other reconnection tests are (because the reconnection logic is all in ScriptNode).